app/jobs/course/course_github_repo_get_commits.rb

```diff
@@ -0,0 +1,199 @@
+        commit.commit_hash =  c[:node][:oid]
+        commit.url =  c[:node][:url]
+        commit.commit_timestamp =  c[:node][:author][:date]
+        commit.github_repo = @github_repo
+        commit.branch = "master"
+        commit.committed_via_web = c[:node][:committedViaWeb]
+      rescue
+        return 0
+      end
+      commit.filenames_changed = filenames_changed(commit.commit_hash)
+      
+      begin  # "try" block
+        uid = c[:node][:author][:user][:databaseId]
+        commit.roster_student = lookup_roster_student_by_github_uid(uid)
+      rescue # optionally: `rescue Exception => ex`
+        uid = ""
+        commit.roster_student = nil
+      end 
+      begin
+        commit.save!
+        return 1
+      rescue
+        return 0
+      end
+    end
+
+    # Regrettably, the v4 graphql api doesn't yet support
+    # getting the filenames changed.  
+    # See: https://github.community/t/get-a-repositorys-commits-along-with-changed-patches-and-the-url-to-changed-files-using-graphql-v4/13585
+    def filenames_changed(commit_sha)
+      begin
+        github_commit_object_api_v3 = github_machine_user.commit(@github_repo_full_name,commit_sha)
+        result = github_commit_object_api_v3[:files].map{ |t| t[:filename]}.to_s
+      rescue
+        result = ""
+      end
+      result
+    end
+
+    def update_one_commit(commit, c)
+      begin
+        commit.files_changed = c[:node][:changedFiles]
+        commit.message = c[:node][:message]
+        commit.commit_hash =  c[:node][:oid]
+        commit.url =  c[:node][:url]
+        commit.commit_timestamp =  c[:node][:author][:date]
+        commit.github_repo = @github_repo
+        commit.branch = "master"
+        commit.committed_via_web = c[:node][:committedViaWeb]
+      rescue
+        return 0
+      end
+
+      commit.filenames_changed = filenames_changed(commit.commit_hash)
+
+      begin  # "try" block
+        uid = c[:node][:author][:user][:databaseId]
+        commit.roster_student = lookup_roster_student_by_github_uid(uid)
+      rescue # optionally: `rescue Exception => ex`
+        uid = ""
+        commit.roster_student = nil
+      end 
+      begin
+        commit.save!
+        return 1
+      rescue
+        return 0
+      end
+    end
+
+    def lookup_roster_student_by_github_uid(uid)
+      @course.student_for_uid(uid)
+    end
+
+    def perform_graphql_query(repo_name, org_name, after="")
+      graphql_query_string = graphql_query(repo_name, org_name, after)
+      github_machine_user.post '/graphql', { query: graphql_query_string }.to_json
+    end
+
+    def graphql_query(repo_name, org_name, after)
+        if after != ""
+          after_clause=", after: \"#{after}\""
+        else
+          after_clause = ""  
+        end
+        # Although the published limit on Graphql queries is 100,
+        # in practice, we've found that it sometimes fails.
+        # 50 seems to be safer round number.
```

Comment:
All I can report is what my experience was.     We had unpredictable flakiness when putting in 100.  Backing off to 50 fixed it.      I know that it "shouldn't fail", but it did.

A lot of the GraphQl stuff is fairly new, so it may have bugs on the GitHub side that are being ironed out.

Reply:
Revisit in issue #299